pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -463,13 +463,12 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 }
 
 func setUpstreamProtocol(cluster *v2.Cluster, port *model.Port) {
-	if port.Protocol.IsHTTP2() {
-		cluster.Http2ProtocolOptions = &core.Http2ProtocolOptions{
```

Comment:
change this as follows:
if port.Protocol is plain HTTP, you can set the use downstream protocol 
else if port.protocol is http2 (user is requesting use of http2), set the http2 protocol options.

Reply:
@rshriram  thanks for the review, made the changes you mentioned above.